Mage.Sets/src/mage/cards/s/SlagstoneRefinery.java

```diff
@@ -59,10 +60,25 @@ public boolean checkEventType(GameEvent event, Game game) {
     @Override
     public boolean checkTrigger(GameEvent event, Game game) {
         ZoneChangeEvent zEvent = (ZoneChangeEvent) event;
-        return zEvent.getFromZone().match(Zone.BATTLEFIELD)
-                && (zEvent.getToZone().match(Zone.GRAVEYARD) || zEvent.getToZone().match(Zone.EXILED))
-                && zEvent.getTargetId().equals(getSourceId())
-                || (zEvent.getTarget().isArtifact(game) && !(zEvent.getTarget() instanceof PermanentToken));
+
+        // Filters events for battlefield => graveyard or exile => graveyard.
+        if (zEvent.getFromZone() == Zone.BATTLEFIELD && (zEvent.getToZone().match(Zone.GRAVEYARD) || zEvent.getToZone().match(Zone.EXILED))) {
```

Comment:
Should be consistent (use either == or .match).  I think == is fine here.  match checks for Zone.ALL but I don't think that's relevant in this context.